### ISSUE TITLE
Clarification: no violation for preceding whitespaces for paragraph tag content

### ISSUE THREAD

**Issue by Zopsss** [checkstyle/checkstyle#15762]
I have read check documentation: https://checkstyle.org/checks/javadoc/javadocparagraph.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

https://checkstyle.org/google_style.html#a7.1.2
https://checkstyle.org/styleguides/google-java-style-20220203/javaguide.html#s7.1.2-javadoc-paragraphs

> Each paragraph except the first has <p> immediately before the first word, with no space after it. 

Detected at: https://github.com/checkstyle/checkstyle/pull/15686#discussion_r1788259499

Following type of code should give violation:
```java
/**
 * Some summary.
 *
 *<p>
 *  <ul> // false-negative
 *    <p> // false-negative
 *      <li>1</li> // false-negative
 *    </p> // false-negative
 *  </ul> // false-negative
 *</p>
 */
public class Test {

    /**
     * Some summary.
     *
     * <p>
     *   <pre>testing...</pre> // false-negative
     *   <pre>testing...</pre> // false-negative
     * </p>
     */
    public void fooooo() {}

    /**
     *  Some summary.
     *
     * <p>
     *  <table> // false-negative
     *  </table> // false-negative
     * </p>
     */
    public void foooo() {}
}
```

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocParagraph" />
  </module>
</module>
```

```
→ java -jar .\checkstyle-10.18.3-SNAPSHOT-all.jar -c .\javadocparagraph_config.xml .\Test.java
Starting audit...
[ERROR] Test.java:4:3: <p> tag should not precede HTML block-tag '<ul>', <p> tag should be removed. [JavadocParagraph]
[ERROR] Test.java:17:8: <p> tag should not precede HTML block-tag '<pre>', <p> tag should be removed. [JavadocParagraph]
[ERROR] Test.java:27:8: <p> tag should not precede HTML block-tag '<table>', <p> tag should be removed. [JavadocParagraph]
Audit done.
Checkstyle ends with 3 errors.
```

Nested content of paragraph tag should not be preceded more than one space.

This rule applies whether `allowNewlineParagraph` is set to false or not.

This issue will be follow up of PR: https://github.com/checkstyle/checkstyle/pull/15686

A new violation message might be required for this.

**Comment by Zopsss** [comment]
> Nested content of paragraph tag should not be preceded more than one space.

one space is allowed right?

**Comment by romani** [comment]
No space should be allowed.
If allowNewlineParagraph=true, we allow \n to be in between.

If you speak about usual space after `*`. It should not be considered as text space, but should be considered as like technical separation from javadoc elements.

**Comment by Anmol202005** [comment]
i'm on it !!

**Comment by romani** [comment]
Hidden knowledge is that this Checks is not very general, and main goal of its existence is to cover Google style requirements. So let's focus on this.

Attention to:

> Each paragraph except the first has immediately before the first word, with no space after it


I think we misinterpreted Google requirements.
Google care about space only because it can cause formatting issue , like extra space (indentation for first sentence of paragraph). Such extra space will be visible in html rendering in browser.
Spaces between html tags is not rendered in browser, such spaces are just for formatting of html code for humans to read, all extra spaces between tags are ignored by browser and has no effect.

So it is very good that we do not put violation on such code that is shown in issue description, it is formatting of html code to let humans to read it clearly.

In scope of this issue let's just remove " until" comments from our input sources.

@Anmol202005 , you PR helped me a lot to rethink this issue.

**Comment by romani** [comment]
Removal of comments "until" is done.

@Anmol202005 , thanks a lot for this.
